llvm/lib/Target/ARM/ARMISelLowering.cpp

```diff
@@ -4617,6 +4636,15 @@ SDValue ARMTargetLowering::LowerFormalArguments(
           (VA.getValVT() == MVT::f16 || VA.getValVT() == MVT::bf16))
         ArgValue = MoveToHPR(dl, DAG, VA.getLocVT(), VA.getValVT(), ArgValue);
 
+      // On CMSE Entry Functions, formal integer arguments whose bitwidth is
```

Comment:
Should we also disable the code above here which inserts `ISD::AssertSext` and `ISD::AssertZext`? If we leave them in, it might be possible for optimisations to prove that the newly-added `TRUNCATE`s are pointless and remove them.

Reply:
The code above regarding SExt and ZExt seems to be dead as it is not code covered by the LLVM tests. The BCvt part is exercised when arguments are of floating-point type. So I believe we can leave it as it is.

Reply:
If it's dead can it be removed? I'm concerned that, if there is a case where it does get executed, then it could cause this truncate-extend pair to be removed, and we'd have the security vulnerability again.